File: src/com/colacelli/irclib/connection/listeners/OnDisconnectListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.connection.Connection;
import com.colacelli.irclib.connection.Server;

public interface OnDisconnectListener extends Listener {
    Type TYPE = Type.DISCONNECT;

    void onDisconnect(Connection connection, Server server);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/com/colacelli/irclib/connection/listeners/OnDisconnectListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.connection.Connection
import com.colacelli.irclib.connection.Server

interface OnDisconnectListener : Listener {
    fun onDisconnect(connection: Connection, server: Server)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.DISCONNECT
    }
}
